src/query/functions/aggregation/quantile_test.go

```diff
@@ -117,6 +117,16 @@ func TestQuantileFnSingleNonNan(t *testing.T) {
 	test.EqualsWithNansWithDelta(t, expected, actual, math.Pow10(-5))
 }
 
+func TestQuantileNanArgument(t *testing.T) {
+	values := []float64{0.0, 1.0}
+	buckets := []int{0, 1}
+
+	expected := math.NaN()
+	actual := bucketedQuantileFn(math.NaN(), values, buckets)
```

Comment:
Also please cover the cases with empty `values` and empty `buckets` (if they are not currently covered).
Can do it in the same test function, especially if you inline those arguments a bit.